botorch/optim/utils/numpy_utils.py

```diff
@@ -125,7 +124,12 @@ def set_tensors_from_ndarray_1d(
             try:
                 size = tnsr.numel()
                 vals = array[index : index + size] if tnsr.ndim else array[index]
-                tnsr.copy_(as_tensor(vals).to(tnsr).view(tnsr.shape).to(tnsr))
+                tnsr.copy_(
+                    torch.as_tensor(vals, device=tnsr.device, dtype=tnsr.dtype)
+                    .to(tnsr)
+                    .view(tnsr.shape)
+                    .to(tnsr)
```

Comment:
These `.to` calls are redundant, since you already specified device and dtype in `as_tensor` call.
```suggestion
                    .view(tnsr.shape)
```

Reply:
Thanks for pointing out :)